### PR TITLE
wip : deployment steps

### PR DESCRIPTION
**Pull request by fbielejec**
This PR add a deploy job to the workflow which is run on every merge to main branch. It will:

- get the build artifact
- build docker image
- push it to ECR (in us-east-2 region for now, where the devnet v1 will reside)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM debian:stretch-slim
+
+WORKDIR node
+
+COPY target/release/aleph-node /node/aleph-node
+COPY docker/run_node.sh /node/run_node.sh
+RUN mkdir /node/tmp/
+
+ENTRYPOINT ["./run_node.sh"]
```
Comment (timorl) [inline docker/Dockerfile:9]: Uh, did anyone check this actually works? Because the lack of complicated `rustup` commands here makes me doubtful.

**Review by timorl — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (fbielejec) [reply docker/Dockerfile:9]: CI builds the binary artifact in a previous step: https://github.com/Cardinal-Cryptography/aleph-node/pull/86/files#diff-28802fbf11c83a2eee09623fb192785e7ca92a3f40602a517c011b947a1822d3R72

Here it's just wrapped as a docker image. Works on my machine :smile:

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -42,46 +45,107 @@ jobs:
...
-# TODO: deploy job
-# -> get the build artifact
-# -> build docker image
-# -> push to ECR in every region
-# -> update ECS tasks in every region
+      - name: Build, tag, and push image to ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          IMAGE_TAG: latest #${{ github.sha }}
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply .github/workflows/deploy.yml]: Fixed

File: docker/Dockerfile (hunk 1 above)
Comment (fbielejec) [reply docker/Dockerfile:9]:
```
  aleph-node:
    image: 436875894086.dkr.ecr.us-east-2.amazonaws.com/aleph-node:latest
    container_name: aleph-node
    expose:
      - 9933
      - 9944
      - 30334
    ports:
      - 9933:9933
      - 9944:9944
      - 30334:30334
    labels:
      - "com.centurylinklabs.watchtower.enable=true"
    volumes:
      - /home/filip/CloudStation/aleph/aleph0/tmp/:/tmp/
    environment:
      - CHAIN_NAME=testnet1
      - BASE_PATH=/tmp/$AUTHORITY
      - NODE_NAME=$AUTHORITY
      - NODE_KEY_PATH=/tmp/$AUTHORITY/libp2p_secret
      - KEY_PATH=/tmp/authorities_keys
      - RESERVED_NODES=/ip4/54.93.81.166/tcp/30334/p2p/12D3KooWQg23osB6yU1BELu6VgMPA8u3efE5r2i9vjHWJZgSawWC /ip4/35.158.209.65/tcp/30334/p2p/12D3KooWJJq9KoG5F8NH9TPc3oje5wVFRnufja28A5NAhc2RTV6v /ip4/18.194.249.135/tcp/30334/p2p/12D3KooWCSnVn3828nxKEaPb5wNB93gE8fcqnZNb9R8Y52ZrRgvi /ip4/18.184.39.81/tcp/30334/p2p/12D3KooWPPdmRAjAAREK1DbRkfPjXJFvYJy15eqjWiG65R4aTnF9 /ip4/18.184.48.43/tcp/30334/p2p/12D3KooWDA5xtT4gky7BHJkLdun4HSKeQVq6xNKj7JkM5CSxmNZe /ip4/3.68.80.96/tcp/30334/p2p/12D3KooWAV1mp8DUdZFhU7NXViAw4ad7XY14W3mD9x3CnxFkHXJy
```

Comment (timorl) [reply docker/Dockerfile:9]: Ah, right!

**Review by timorl — APPROVED** [review]
LGTM, but I'm not particularly confident in my understanding of this.

**Review by michalseweryn — APPROVED** [review]

**Review by mike1729 — APPROVED** [review]

**Pull request merged**